src/main/features/ccj/form/models/payBySetDate.ts

```diff
@@ -0,0 +1,46 @@
+import { IsDefined, ValidateNested } from 'class-validator'
+
+import { IsValidLocalDate } from 'forms/validation/validators/isValidLocalDate'
+import { IsTodayOrInFuture } from 'forms/validation/validators/IsTodayOrInFuture'
+
+import { Serializable } from 'models/serializable'
+import { LocalDate } from 'forms/models/localDate'
+import { isValidYearFormat } from 'app/forms/validation/validators/isValidYearFormat'
+
+export class ValidationErrors {
+  static readonly DATE_REQUIRED: string = 'Enter a date'
+  static readonly DATE_NOT_VALID: string = 'Enter a valid date'
+  static readonly DATE_TODAY_OR_IN_FUTURE: string = 'Enter a date that is today or in the future'
+  static readonly DATE_INVALID_YEAR: string = 'Enter a 4 digit year'
+
+}
+
+export class PayBySetDate implements Serializable <PayBySetDate> {
+
+  @ValidateNested()
+  @IsDefined({ message: ValidationErrors.DATE_REQUIRED })
+  @isValidYearFormat(4, { message: ValidationErrors.DATE_INVALID_YEAR })
```

Comment:
not covered in unit tests

Reply:
Please see src/main/app/forms/models/address.ts:21 ValidationConstants
for constants used in validation annotations.